gsql/cache/cache_key.py

```diff
@@ -0,0 +1,74 @@
+from functools import lru_cache
+from key_pair import Keypair
+from gsql.cache.cache_handler import (
+    CacheDict,
+    check_json_file_exists,
+    count_lines_in_json,
+    create_json_file,
+    keys_exists,
+    read_from_json,
+    remove_key_from_json,
+    write_to_json,
+)
+from gsql.cache.constant import MAXIMUM_SIZE_OF_JSON_FILE
+
+
+"""
+    Cachekey contains the type of query,
+    the query itself, database id and the sheet id.
+"""
+
+
+class Cachekey:
+    def __init__(self, query_type, query, db_id, sheet_id) -> None:
+        self.query_type = query_type
+        self.query = query
+        self.db_id = db_id
+        self.sheet_id = sheet_id
+
```

Comment:
##  Design Issue

What happens when in case of a nested query like
```
SELECT NAME FROM CUSTOMERS WHERE ID  IN  (SELECT ID FROM CUSTOMER_ORDERS); 
```

which sheet_id to consider as there are two tables associated - CUSTOMERS, CUSTOMER_ORDERS
We can always take the first table(sheet_id) but let's give this a thought.

Reply:
Kool